rpcserver.go

```diff
@@ -970,6 +974,111 @@ func (r *rpcServer) EstimateFee(ctx context.Context,
 	return resp, nil
 }
 
+// unmarshallOutPoint converts an outpoint from its lnrpc type to its canonical
+// type.
+func unmarshallOutPoint(op *lnrpc.OutPoint) (*wire.OutPoint, error) {
+	var hash chainhash.Hash
+	switch {
+	case len(op.TxidBytes) == 0 && len(op.TxidStr) == 0:
+		fallthrough
+
+	case len(op.TxidBytes) != 0 && len(op.TxidStr) != 0:
+		return nil, fmt.Errorf("either TxidBytes or TxidStr must be " +
+			"specified, but not both")
+
+	// The hash was provided as raw bytes.
+	case len(op.TxidBytes) != 0:
+		copy(hash[:], op.TxidBytes)
+
+	// The hash was provided as a hex-encoded string.
+	case len(op.TxidStr) != 0:
+		h, err := chainhash.NewHashFromStr(op.TxidStr)
+		if err != nil {
+			return nil, err
+		}
+		hash = *h
+	}
+
+	return &wire.OutPoint{
+		Hash:  hash,
+		Index: op.OutputIndex,
+	}, nil
+}
+
+// BumpFee allows bumping the fee rate of an arbitrary input. The fee rate can
+// be expressed either as a specific fee rate or a delta of blocks in which the
+// output should be swept on-chain within. If a fee rate is not explicitly
+// specified, then a fee rate corresponding to a confirmation target of 6 blocks
+// will be used. The status of the input sweep can be checked through the
+// PendingSweeps RPC.
+func (r *rpcServer) BumpFee(ctx context.Context,
+	in *lnrpc.BumpFeeRequest) (*lnrpc.BumpFeeResponse, error) {
+
+	// Parse the outpoint from the request.
+	op, err := unmarshallOutPoint(in.Outpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	// Construct the request's fee preference.
+	satPerKw := lnwallet.SatPerKVByte(in.SatPerByte * 1000).FeePerKWeight()
+	feePreference := sweep.FeePreference{
+		ConfTarget: uint32(in.TargetConf),
+		FeeRate:    satPerKw,
+	}
+
+	// Before attempting to bump the fee of the output, we'll first attempt
+	// to retrieve it from our wallet.
+	txOut, err := r.server.cc.wallet.FetchInputInfo(op)
+	switch {
+	// It's possible for ErrNotMine to be returned if the output exists
+	// within the wallet, but it is not an output for which we have control
+	// of, e.g., CommitmentTimeLock, CommitmentNoDelay outputs, so we'll
+	// still attempt to send it over to the UtxoSweeper to see if it's
+	// currently attempting to sweep it.
+	case err == lnwallet.ErrNotMine:
```

Comment:
How does this compare to first checking if the sweeper knows it (now that it exposes its inputs) and only if it doesn't, check the wallet? Are there other wallet inputs that we have no control over but are also not in the sweeper (utxonursery maybe?)

Reply:
> How does this compare to first checking if the sweeper knows it (now that it exposes its inputs) and only if it doesn't, check the wallet?

Same thing really, I opted to do it this way to prevent more code within the sweeper.

> Are there other wallet inputs that we have no control over but are also not in the sweeper (utxonursery maybe?)

The wallet only has the keys for addresses it's derived. If the input exists within the nursery, then that would mean it's not ready to be swept yet, no? So there'd be no point in attempting to bump its fee.

Reply:
Updated so that it checks the sweeper for the input first.